LeaderboardBackend.Test/Leaderboards.cs

```diff
@@ -339,6 +341,71 @@ public async Task DeletedBoardsDontConsumeSlugs()
         created!.CreatedAt.Should().Be(_clock.GetCurrentInstant());
     }
 
+    [Test]
+    public async Task RestoreLeaderboard_OK()
+    {
+        ApplicationContext context = _factory.Services.CreateScope().ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        Leaderboard deletedBoard = new()
+        {
+            Name = "Super Mario World",
+            Slug = "super-mario-world-deleted",
+            DeletedAt = _clock.GetCurrentInstant()
+        };
+
+        context.Leaderboards.Add(deletedBoard);
+        await context.SaveChangesAsync();
+        deletedBoard.Id.Should().NotBe(default);
+
+        HttpResponseMessage res = await _apiClient.Put($"/leaderboard/{deletedBoard.Id}/restore", new()
+        {
+            Jwt = _jwt
+        });
+
+        res.StatusCode.Should().Be(HttpStatusCode.NoContent);
+
+        Leaderboard? board = await context.Leaderboards.FindAsync(deletedBoard.Id);
+        board.Should().NotBeNull();
+        // TODO: `DeletedAt` is still not null here. Don't know how to fix it.
```

Comment:
I'm learning a bit too, and apparently when you want to test an assertion, you must pull an entirely new database context - ie:
```cs
        ApplicationContext verifyContext = _factory.Services.CreateScope().ServiceProvider.GetRequiredService<ApplicationContext>();
        Leaderboard? board = await verifyContext.Leaderboards.FindAsync(deletedBoard.Id);
```

Reply:
Thankfully we can just clear the change tracker as Ted's said in the chat. Much smaller word salad too. I've updated.